### PR TITLE
drivers: sensor: SHT2X Added support

### PR DESCRIPTION
**Pull request by panicking**
This adds support for Sensirion's SHT2X temperature and humidity sensor.

### CODE REVIEW

File: drivers/sensor/sht2x/sht2x.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+	return 0;
+}
+
+static int sht2x_init(const struct device *dev)
+{
+	const struct sht2x_config *cfg = dev->config;
+	int rc = 0;
+
+	if (!device_is_ready(cfg->bus.bus)) {
```
Comment (MaureenHelm) [inline drivers/sensor/sht2x/sht2x.c:139]: Use `i2c_is_ready_dt`

File: drivers/sensor/sht2x/sht2x.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+#define SHT2X_INIT(n)						\
+	static struct sht2x_data sht2x_data_##n;		\
+								\
+	static const struct sht2x_config sht2x_config_##n = {	\
+		.bus = I2C_DT_SPEC_INST_GET(n),			\
+	};							\
+								\
+	DEVICE_DT_INST_DEFINE(n,				\
```
Comment (MaureenHelm) [inline drivers/sensor/sht2x/sht2x.c:168]: Use `SENSOR_DEVICE_DT_INST_DEFINE`

File: include/zephyr/drivers/sensor/sht2x.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#include <zephyr/drivers/sensor.h>
+
+/**
+ * @brief Fetches data using the on-chip heater.
+ *
+ * @param dev Pointer to the sensor device
+ *
+ * @return 0 if successful, negative errno code if failure.
+ */
+int sht2x_fetch_with_heater(const struct device *dev);
```
Comment (MaureenHelm) [inline include/zephyr/drivers/sensor/sht2x.h:33]: This isn't implemented anywhere

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensor`

**Comment by str4t0m** [comment]
Thanks for this contribution, but there is not really a need to add a completely new driver for these sensors.
Please have a look at the existing si7006 driver, https://github.com/zephyrproject-rtos/zephyr/pull/56227, and https://github.com/zephyrproject-rtos/zephyr/pull/22862#pullrequestreview-359397725.

**Comment by xyzzy42** [comment]
I have had a PR out for nearly a year, waiting, that adds support without adding a new driver.

**Pull request closed without merging**